File: csharp/Day1.cs
namespace AdventOfCode2024;

public class Day1
{
    public static void Run()
    {
        Console.WriteLine("Hello, World!");
        List<int> left = new List<int>();
        List<int> right = new List<int>();
        int count = 0;
        File.ReadLines("../../../../csharp/day1.txt").ToList().ForEach(x => x.Split("   ").ToList().ForEach(x => { if (count % 2 == 0) {left.Add(int.Parse(x));} else {right.Add(int.Parse(x));} count++; }));

        left.Sort();
        right.Sort();

        Dictionary<int, int> occurances = new Dictionary<int, int>();
        foreach (var x in right.GroupBy(x => x))
        {
            occurances[x.Key] = x.Count();
        }

        int sol1 = 0;
        int sol2 = 0;
        for (int i = 0; i < left.Count; i++)
        {
            sol1 += Math.Abs(left[i] - right[i]);
            if (occurances.ContainsKey(left[i]))
            {
                sol2 += occurances[left[i]] * left[i];
            }
        }

        Console.WriteLine(sol1);
        Console.WriteLine(sol2);
    }
}

File: python/day1.py
from collections import Counter

file = open('day1.txt', 'r')
lines = file.readlines()

left = []
right = []
part1 = 0
part2 = 0

for line in lines:
    tmp = line.strip().split('   ')
    left.append(int(tmp[0]))
    right.append(int(tmp[1]))

rightOcc = Counter(right)
for num in left:
    part2 += num * rightOcc[num]
    
left.sort()
right.sort()

for index in range(0, len(left)):
    result = left[index] - right[index]
    if result < 0: part1 -= result
    else: part1 += result
        
print(part1)
print(part2)